test/functional/product/list_test.rb

```diff
@@ -0,0 +1,78 @@
+require File.join(File.dirname(__FILE__), '../test_helper')
+require File.join(File.dirname(__FILE__), 'product_helpers')
+
+describe "listing products" do
+  include ProductHelpers
+
+  before do
+    @cmd = %w(product list)
+  end
+
+  let(:org_id) { 1 }
+  let(:subscription_id) { 1 }
+  let(:sync_plan_id) { 1 }
+  let(:empty_response) do
+    {
+      "total" => 0,
+      "subtotal" => 0,
+      "page" => "1",
+      "per_page" => "1000",
+      "error" => nil,
+      "search" => nil,
+      "sort" => {
+        "by" => nil,
+        "order" => nil
+      },
+      "results" => []
+    }
```

Comment:
You can use `index_response` from [hammer-cli-foreman](https://github.com/theforeman/hammer-cli-foreman/blob/master/lib/hammer_cli_foreman/testing/api_expectations.rb#L53)